frame/grandpa/src/lib.rs

```diff
@@ -359,6 +452,22 @@ impl<T: Trait> Module<T> {
 			Self::set_grandpa_authorities(authorities);
 		}
 	}
+
+	/// Submits an extrinsic to report an equivocation. This method will sign an
+	/// extrinsic with a call to `report_equivocation` with any reporting keys
+	/// available in the keystore and will push the transaction to the pool.
+	/// Only useful in an offchain context.
+	pub fn submit_report_equivocation_extrinsic(
+		equivocation_proof: EquivocationProof<T::Hash, T::BlockNumber>,
+		key_owner_proof: <T::HandleEquivocation as equivocation::HandleEquivocation<T>>::KeyOwnerProof,
```

Comment:
We would have to parameterize `EquivocationProof` on `K` (`K = <T::HandleEquivocation as equivocation::HandleEquivocation<T>>::KeyOwnerProof`), which would make it impossible to define the runtime API since we don't have access to that type there (https://github.com/paritytech/substrate/pull/3868/files#diff-233fff7c89a6b4ca96ca4ee31c62088cR483).